src/test/ui/feature-gates/feature-gate-cfg-version.rs

```diff
@@ -0,0 +1,40 @@
+#[cfg(version("1.44"))]
+//~^ ERROR `cfg(version)` is experimental and subject to change
+fn foo() -> bool { true }
+#[cfg(not(version("1.44")))]
+//~^ ERROR `cfg(version)` is experimental and subject to change
+fn foo() -> bool { false }
+
+#[cfg(version("1.43", "1.44", "1.45"))] //~ ERROR: expected single version literal
+//~^ ERROR `cfg(version)` is experimental and subject to change
+fn bar() -> bool  { false }
+#[cfg(version(false))] //~ ERROR: expected a version literal
+//~^ ERROR `cfg(version)` is experimental and subject to change
+fn bar() -> bool  { false }
+#[cfg(version("foo"))] //~ ERROR: invalid version literal
+//~^ ERROR `cfg(version)` is experimental and subject to change
+fn bar() -> bool  { false }
+#[cfg(version("999"))]
+//~^ ERROR `cfg(version)` is experimental and subject to change
+fn bar() -> bool  { false }
+#[cfg(version("-1"))] //~ ERROR: invalid version literal
+//~^ ERROR `cfg(version)` is experimental and subject to change
+fn bar() -> bool  { false }
+#[cfg(version("65536"))] //~ ERROR: invalid version literal
+//~^ ERROR `cfg(version)` is experimental and subject to change
+fn bar() -> bool  { false }
+#[cfg(version("0"))]
+//~^ ERROR `cfg(version)` is experimental and subject to change
+fn bar() -> bool { true }
+
+#[cfg(version("1.65536.2"))]
+fn version_check_bug() {}
```

Comment:
Wow, do we need this? Or maybe we can try another crate `semver`?

Reply:
I think this it's unlikely for anyone to run into this, but I wanted to explicitly document this problem. I have thought about using `semver`, however if I were to use `semver::Version` I'd need to make sure the version contains major, minor and patch versions, and if I were to use `semver::VersionReq` I'd need to check if the version literal doesn't contain things like `>=1.0`, `<1.5`, `>1.2, <1.3` which the RFC only mentions as possible future work ([`However, as a first iteration, version(1.27.0) is simple and covers most use cases.`](https://github.com/rust-lang/rfcs/blob/master/text/2523-cfg-path-version.md#the-bikeshed---attribute-name)). Unless it's okay to implement this using `VersionReq` and adding support for `>`, `<`, etc. syntax I think it's simpler to just use `version_check` despite this bug which I believe is unlikely to affect anyone.